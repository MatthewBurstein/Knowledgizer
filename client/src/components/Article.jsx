import React, { Component } from 'react';
import { wikiGetRequest } from '../helpers/requestHelper';

class Article extends Component {
  constructor() {
    super();

    let article =  wikiGetRequest();

    this.state = article;
  }


  render() {
    return(
      <div>
        <a href={this.state.url}>
          {this.state.title}
        </a>
      </div>
    )
  }
}

export default Article;
