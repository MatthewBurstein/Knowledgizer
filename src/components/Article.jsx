import React, { Component } from 'react';

class Article extends Component {
  constructor() {
    super();

    let article =  { title: 'some headline', body: 'some body', url: 'some url'}

    this.state = article;
  }


  render() {
    return(
      <div>
        <a href={this.state.url}>
          {this.state.title}
        </a>
      </div>
    )
  }
}

export default Article;
