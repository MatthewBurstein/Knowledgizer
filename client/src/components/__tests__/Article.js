import React from 'react';
import Article from '../Article';
import { shallow } from 'enzyme';

describe('Article', () => {
  const article = shallow(<Article />)
  it('renders properly', () => {
    expect(article).toMatchSnapshot();
  });

  it('intializes an article with a title in state', () => {
    expect(article.state().title).toBeDefined;
  });
});
