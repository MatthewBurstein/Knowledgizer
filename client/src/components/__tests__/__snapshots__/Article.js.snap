// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Article />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <a
        href="some url"
>
        some headline
</a>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "some headline",
        "href": "some url",
      },
      "ref": null,
      "rendered": "some headline",
      "type": "a",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <a
          href="some url"
>
          some headline
</a>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "some headline",
          "href": "some url",
        },
        "ref": null,
        "rendered": "some headline",
        "type": "a",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
